### PR TITLE
Marin/conversation datasource refactor

### PR DESCRIPTION
**Pull request by MartinJibber**
Found a way for conversations and thread to share the same data source.

### CODE REVIEW

File: iOS/Flows/Conversation/ConversationViewControllerDataSource.swift [hunk 1]
```diff
@@ -66,7 +66,7 @@ class ConversationCollectionViewDataSource: CollectionViewDataSource<Conversatio
 
     /// Updates the datasource with the passed in array of message changes.
     func update(with changes: [ListChange<Message>],
-                conversationController: ConversationController,
+                conversationController: ConversationControllerType,
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/ConversationViewControllerDataSource.swift:69]: A common interface for message and conversation controllers.

File: iOS/Flows/Conversation/ConversationViewControllerDataSource.swift [hunk 2]
```diff
@@ -55,7 +55,7 @@ class ConversationCollectionViewDataSource: CollectionViewDataSource<Conversatio
                                                                     item: (channelID, messageID, self))
 
             let interaction = UIContextMenuInteraction(delegate: self.contextMenuDelegate)
-            cell.addInteraction(interaction)
+            cell.contentView.addInteraction(interaction)
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/ConversationViewControllerDataSource.swift:58]: This is still wrong, but no worse than before.

File: iOS/Flows/Conversation/ConversationViewControllerDataSource.swift (hunk 1 above)
Comment (Benjamindodgson) [reply iOS/Flows/Conversation/ConversationViewControllerDataSource.swift:69]: yep. This makes total sense.

**Review by Benjamindodgson — APPROVED** [review]

File: iOS/Flows/Conversation/ConversationViewControllerDataSource.swift (hunk 2 above)
Comment (MartinJibber) [reply iOS/Flows/Conversation/ConversationViewControllerDataSource.swift:58]: @Benjamindodgson Agreed, it's just left over from a test I was doing. But doesn't make a difference.

**Pull request merged**